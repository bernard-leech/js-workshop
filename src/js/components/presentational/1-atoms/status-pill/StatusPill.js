// @flow
import React from 'react'
import classNames from 'classnames'

// import type { StatusPillType } from './StatusPill.types'

function StatusPill (props) {
  const {
    status,
    children
  } = props

  const statusPillClasses = classNames('status-pill',
    {
      'status-pill--error': status === 'error',
      'status-pill--success': status === 'success'
    }
  )

  return (
    <div className={statusPillClasses}>
      {children}
    </div>
  )
}

export default StatusPill
