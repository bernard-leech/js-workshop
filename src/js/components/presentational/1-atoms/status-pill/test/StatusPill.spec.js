import React from 'react'
import TestRenderer from 'react-test-renderer'

import { propsError, propsSuccess } from './StatusPill.mock'

import StatusPill from '../StatusPill'

it('renders correctly', () => {
  const tree = TestRenderer
    .create(<StatusPill />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly - success', () => {
  const tree = TestRenderer
    .create(<StatusPill {...propsSuccess} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly - error', () => {
  const tree = TestRenderer
    .create(<StatusPill {...propsError} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
