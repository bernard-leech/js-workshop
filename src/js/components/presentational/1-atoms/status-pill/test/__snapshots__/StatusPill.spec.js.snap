// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly - error 1`] = `
<div
  className="status-pill status-pill--error"
>
  Overdue
</div>
`;

exports[`renders correctly - success 1`] = `
<div
  className="status-pill status-pill--success"
>
  Paid
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="status-pill"
/>
`;
